pyfolio/sanitize.py

```diff
@@ -0,0 +1,237 @@
+        - If not, localize to UTC and warn
+    - There must a column called 'cash'
+        - If not, raise ValueError
+    - Positions must be floats
+        - if not, coerce values to float and warn
+        - Failing that, raise a ValueError
+    """
+    if not isinstance(positions, pd.DataFrame):
+        msg = '`positions` is not a pd.DataFrame.'
+        raise ValueError(msg)
+
+    if positions.isnull().any().any():
+        positions = positions.fillna(0)
+        msg = '`positions` contains NaNs. Filling with 0...'
+        warn(msg)
+
+    if positions.index.tz is None:
+        positions.index = positions.index.tz_localize('UTC')
+        msg = ('`positions` index is not timezone-localized. '
+               'Localizing to UTC...')
+        warn(msg)
+
+    if 'cash' not in positions.columns:
+        msg = '`positions does not contain a `cash` column.'
+        raise ValueError(msg)
+
+    if not all(positions.apply(lambda col: col.dtype,
+                               axis='columns') == float):
+        try:
+            positions = positions.astype(float)
+            msg = '`positions` does not have float dtype. Coercing to float...'
+            warn(msg)
+        except ValueError:
+            msg = ('`positions` does not have float dtype, and could not be '
+                   'coerced into floats.')
+            raise ValueError(msg)
+
+    return positions
+
+
+def sanitize_txns(txns):
+    """
+    Sanitize transactions.
+
+    Returns
+    -------
+    txns : pd.DataFrame
+        Executed trade volumes and fill prices.
+        - One row per trade.
+        - Trades on different names that occur at the
+          same time will have identical indicies.
+        - Example:
+            index                  amount   price    symbol
+            2004-01-09 12:18:01    483      324.12   'AAPL'
+            2004-01-09 12:18:01    122      83.10    'MSFT'
+            2004-01-13 14:12:23    -75      340.43   'AAPL'
+
+    Notes
+    -----
+    - Must be a DataFrame
+        - If not, raise ValueError
+    - There must not be any NaNs
+        - If there are, raise a ValueError
+    - Index must be timezone-localized datetimes
+        - If not, localize to UTC and warn
+    - There must be columns called 'amount', 'price', and 'symbol'
+        - If not, raise a ValueError
+    - Amounts must be ints
+    - Prices must be floats
+    - Symbols must be strings or ints
+        - If not, warn
+    """
+    if not isinstance(txns, pd.DataFrame):
+        msg = '`txns` is not a pd.DataFrame.'
+        raise ValueError(msg)
+
+    if txns.isnull().any().any():
+        msg = '`txns` contains NaNs.'
+        raise ValueError(msg)
+
+    if txns.index.tz is None:
+        txns.index = txns.index.tz_localize('UTC')
+        msg = ('`txns` index is not timezone-localized. '
+               'Localizing to UTC...')
+        warn(msg)
+
+    if not set(['amount', 'price', 'symbol']) <= set(txns.columns):
```

Comment:
An idiom I've used for this kind of check a bunch of times is:

```python
expected = {'amount', 'price', 'symbol'}
received = set(txns.columns)
missing = received - expected
# This is useful to include b/c it often catches typos, e.g., "smybol" instead
# of "symbol"
unexpected = expected - received
if missing:  # sometimes it's also an error to pass unexpected columns.
    raise format_nice_error_message(expected, received, missing, unexpected)
```

Reply:
Clever! I like it.